### ISSUE TITLE
Add a dashboard

### ISSUE THREAD

**Issue by cbraillon** [nunomaduro/phpinsights#232]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes
| Library version  | latest

It would be great to see results in a dashboard.
I know that [Symfony Insight](https://insight.symfony.com/) has one and it's awesome.

It will increase the readability of issues.

<!--
Love PHP Insights? Please consider supporting us via Patreon or OpenCollective:
👉  https://www.patreon.com/nunomaduro
👉  https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=66BYDWAT92N6L
--

**Comment by caneco** [comment]
I wouldn't mind to tackle this… 🔥 🔥 🔥

**Comment by olivernybroe** [comment]
We have a PR for adding formatters #201, which will make doing this a lot easier.

I see two ways of implementing this. Either create a new formatter called `html`, which generates a site from the output. Or create a site which can load the data from the json format.

**Comment by caneco** [comment]
@olivernybroe in each run we could add a new JSON file to a specific folder that a HTML would read an present the key figures and charts.

Feels like the best approach, and also to show the progress of the insights results.

**Comment by olivernybroe** [comment]
@caneco Sounds like a good idea. I think the JSON way is more robust and extendable in the future.

Would also do so you could compare your results for each run 👍 

Maybe this dashboard should be a package for itself that you can just `composer require` in?

**Comment by caneco** [comment]
Could be something installed if needed with `npm run install && npm run build/serve`?

**Comment by olivernybroe** [comment]
@caneco Yeah, it makes a ton of sense to use `npm` instead as it is front end stuff.

**Comment by caneco** [comment]
Another question…

Should the dashboard be:
- only available by running a local server (`npm run serve`)
- available as a "static" website in the repo (`npm run build`)
- or… both

**Comment by olivernybroe** [comment]
I think a static site might be the best choice, if that is not too much work. Else just do what you find easiest.

**Comment by cbraillon** [comment]
Love the idea of a static site !

**Comment by nunomaduro** [comment]
Thanks to @olivernybroe, we can know get `json` format from phpinsights.

**Comment by Gummibeer** [comment]
I also would like static html files more. And at the end it's just adding tag wrappers around the already present text format?
`<dl>`, `<ul>` and this kind of stuff should do the trick?

**Comment by Gummibeer** [comment]
Will try to add a HTML formatter which generates a static file.

**Comment by Gummibeer** [comment]
I'm 💯 % not in design! But this is my first/working result. I've tried to rebuild the console output - because I'm a lazy-bitch I use bootstrap. 🙈 
My idea is that anyone else can make it pretty if wanted, but for the moment it should be fine!?

![screencapture-localhost-63342-phpinsights-dashboard-html-2019-08-28-14_58_11](https://user-images.githubusercontent.com/6187884/63857606-4de23280-c9a4-11e9-983e-574973054f77.png)

**Comment by nunomaduro** [comment]
@caneco

**Comment by Wulfheart** [comment]
I am not sure if this is already present. I can't see it in the docs.
